libs/SalesforceSDKCore/SalesforceSDKCore/Classes/KeyValueEncryptedFileStore.swift

```diff
@@ -0,0 +1,395 @@
+            }
+            guard FileManager.default.fileExists(atPath: fileURL.path) else {
+                return nil
+            }
+
+            do {
+                let encryptedData = try Data(contentsOf: fileURL)
+                guard let decryptedData = encryptionKey.decryptData(encryptedData) else {
+                    SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to decrypt file at path '\(fileURL.path)'")
+                    return nil
+                }
+                return String(data: decryptedData, encoding: .utf8)
+            } catch {
+                SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Error reading file at path '\(fileURL.path)': \(error)")
+                return nil
+            }
+        }
+
+        set {
+            if let newValue = newValue {
+                saveValue(newValue, forKey: key)
+            } else {
+                removeValue(forKey: key)
+            }
+        }
+    }
+    /// Removes entry for given key.
+    /// - Parameter key: The key associated with the entry to remove.
+    /// - Returns: True if the entry is successfully removed or doesn't exist, false otherwise.
+    @objc @discardableResult public func removeValue(forKey key: String) -> Bool {
+        guard let fileURL = encodedUrl(forKey: key, function: #function) else {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to construct file URL")
+            return false
+        }
+        guard FileManager.default.fileExists(atPath: fileURL.path) else {
+            return true
+        }
+
+        do {
+            try FileManager.default.removeItem(at: fileURL)
+            return true
+        } catch {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Error removing file at path '\(fileURL.path)': \(error)")
+            return false
+        }
+    }
+
+    /// Removes all contents of the store.
+    @objc public func removeAll() {
+        let files = KeyValueEncryptedFileStore.contentsOfDirectory(directory.path, function: #function)
+        for file in files {
+            let fileURL = directory.appendingPathComponent(file)
+            KeyValueEncryptedFileStore.removeFile(fileURL, function: #function)
+        }
+    }
+
+    /// - Returns: The number of entries in the store.
+    @objc public func count() -> Int {
+        let files = KeyValueEncryptedFileStore.contentsOfDirectory(directory.path, function: #function)
+        return files.count
+    }
+
+    /// - Returns: A Boolean value that indicates whether the store is empty.
+    @objc public func isEmpty() -> Bool {
+        return count() == 0
+    }
+
+    // MARK: Private
+    private static func storesDirectory(forUser user: UserAccount) -> String? {
+        return SFDirectoryManager.shared().directory(forUser: user, type: .libraryDirectory, components: [keyValueStoresDirectory])
```

Comment:
Did library to match sapp, not sure if we want to make it customizable at the shared instance level

Reply:
Let's line this up with the directory we use for `SmartStore`.